action/protocol/staking/handler_stake_migrate.go

```diff
@@ -0,0 +1,181 @@
+package staking
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	accountutil "github.com/iotexproject/iotex-core/action/protocol/account/util"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/state"
+)
+
+const (
+	executionProtocolID = "smart_contract"
+)
+
+type (
+	executionProtocol interface {
+		Handle(ctx context.Context, act action.Action, sm protocol.StateManager) (*action.Receipt, error)
+	}
+)
+
+func (p *Protocol) handleStakeMigrate(ctx context.Context, act *action.MigrateStake, csm CandidateStateManager) ([]*action.Log, []*action.TransactionLog, uint64, error) {
+	actLogs := make([]*action.Log, 0)
+	transferLogs := make([]*action.TransactionLog, 0)
+	si := csm.SM().Snapshot()
+	revertSM := func() {
+		if revertErr := csm.SM().Revert(si); revertErr != nil {
+			log.L().Panic("failed to revert state", zap.Error(revertErr))
+		}
+	}
+
+	// validate bucket index
+	bucket, rErr := p.fetchBucket(csm, act.BucketIndex())
+	if rErr != nil {
+		return nil, nil, 0, rErr
+	}
+	if err := p.validateStakeMigrate(ctx, bucket, csm); err != nil {
+		return nil, nil, 0, err
+	}
+
+	// force-withdraw native bucket
+	staker, rerr := fetchCaller(ctx, csm, big.NewInt(0))
+	if rerr != nil {
+		return nil, nil, 0, errors.Wrap(rerr, "failed to fetch caller")
+	}
+	candidate := csm.GetByIdentifier(bucket.Candidate)
+	if candidate == nil {
+		return nil, nil, 0, errCandNotExist
+	}
+	actLog, tLog, err := p.withdrawBucket(ctx, staker, bucket, candidate, csm)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	actLogs = append(actLogs, actLog.Build(ctx, nil))
+	transferLogs = append(transferLogs, tLog)
+
+	// call staking contract to stake
+	duration := int64(bucket.StakedDuration / p.getBlockInterval(protocol.MustGetBlockCtx(ctx).BlockHeight))
+	excReceipt, err := p.createNFTBucket(ctx, act, bucket.StakedAmount, big.NewInt(duration), candidate, csm.SM())
+	if err != nil {
+		revertSM()
+		return nil, nil, 0, errors.Wrap(err, "failed to handle execution action")
+	}
+	if excReceipt.Status != uint64(iotextypes.ReceiptStatus_Success) {
+		// TODO: return err or handle error?
+		revertSM()
+		return nil, nil, 0, errors.Errorf("execution failed with status %d", excReceipt.Status)
+	}
+	// add sub-receipts logs
+	actLogs = append(actLogs, excReceipt.Logs()...)
+	transferLogs = append(transferLogs, excReceipt.TransactionLogs()...)
+	return actLogs, transferLogs, excReceipt.GasConsumed, nil
```

Comment:
`gasConsumed + 10000`, 10000 is the base fee to withdraw a native bucket

Reply:
fixed